binary/src/Pos/Binary/Class/TH.hs

```diff
@@ -368,6 +368,12 @@ deriveSimpleBiInternal predsMB headTy constrs = do
                 when (length realFields /= length dcFields) $ templateHaskellError $
                     sformat ("Some field of "%shown
                     %" constructor doesn't have an explicit name") cName
+                when (length constrs > 1 && not (null realFields)) $
```

Comment:
If it's a sum type, and any of the constructors take values, it could have partial field accessors.

Reply:
Cool! Doing this was a good idea 👍